### ISSUE TITLE
test sagews utf8 strings

### ISSUE THREAD

**Issue by haraldschilly** [sagemathinc/smc#1859]
python2 is really bad at this. add test cases for smc_sagews to check if evaluating plain unicode strings work. the challenge might be to really test the utf8 strings and not their encoded representation.

i.e. cells like

* `u"ááá"`
* `"ááá"`
* `"öäß"` → `'\xc3\xb6\xc3\xa4\xc3\x9f'`
* `u"öäß"` → `u'\xf6\xe4\xdf'`

**Comment by haraldschilly** [comment]
@DrXyzzy I've pushed https://github.com/sagemathinc/smc/commit/391bfae9bbe305f1080ec385af6831981d6905ba as a new branch https://github.com/sagemathinc/smc/tree/issue/1859 to show you my failed attempt at trying to do this. Several other errors start to pop up when the encoding is declared to be utf8. So maybe one has to create a separate test file just for that ...

**Comment by haraldschilly** [comment]
another idea, maybe the string to test needs to be `'print "äöüß"'` or `u'print "äöüß"'`...

**Comment by DrXyzzy** [comment]
First, I think there is a bug in sagews. This code runs correctly on the command line under 'python2' and 'sage' but shows an error in sagews:
```
for x in [ u"ááá", "ááá", "öäß", u"öäß"]:
    print(x)
-- output: --
ááá
Error in lines 1-2
Traceback (most recent call last):
  File "/projects/ccd4d4a4-29a8-4c39-85c2-a630cb1e9b6c/.local/lib/python2.7/site-packages/smc_sagews/sage_server.py", line 982, in execute
    exec compile(block+'\n', '', 'single') in namespace, locals
  File "", line 2, in <module>
  File "/projects/ccd4d4a4-29a8-4c39-85c2-a630cb1e9b6c/.local/lib/python2.7/site-packages/smc_sagews/sage_server.py", line 403, in write
    self._buf += output.replace('\x00','')
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 0: ordinal not in range(128)
*** WARNING: Code contains non-ascii characters ***
```

Workaround is to catch `UnicodeDecodeError` in call to `output.replace()` and decode output. I'm working on PR now.

**Comment by DrXyzzy** [comment]
I made issues #1865 and #1866 for bugs found while investigating.

process: For the pytest suite for sagews, if the code to be tested is a unicode object, e.g. `u"xxx"`, then pass it to test routines enclosed in single quotes, e.g. `'u"xxx"'`. Doing this causes the exact same message to be passed to sage_server as input from a manually created sagews cell without the single quotes. Unicode testing is complicated by the fact that creating debug output may change representation of the object under inspection.

**Comment by DrXyzzy** [comment]
Bug opened for above: #1865. See also #1866 for another encoding-related bug.

**Comment by DrXyzzy** [comment]
Fixed in #1877
